prediction/independent_poisson_model.py

```diff
@@ -46,11 +47,124 @@ def predict(self, home_team, away_team):
 
         l_A = (uA_B + vB_A) / 2.0
         l_B = (uB_A + vA_B) / 2.0
-
+        # print("l_A,l_B:",l_A,l_B)
         score_dict = self.run_simulations(l_A, l_B, self.NUM_ITERS)
-
+        # print("score_dict",score_dict)
         return score_dict
 
+
+    def getExpectation(self,home_team,away_team):
```

Comment:
the logic here is a replication of code right above it. refactor.

Reply:
This class is also a bottleneck because it calls "filter_matches" (which essentially gets its data from the DB) unnecessarily over and over.